### PR TITLE
Refactor CMake

### PR DESCRIPTION
**Pull request by phmkopp**
Better CMake :)

### CODE REVIEW

**Comment by phmkopp** [comment]
@philbucher I hope the CI stuff still does the same thing as before. Would be nice if you could have a quick look.

**Comment by phmkopp** [comment]
@philbucher Also, I removed the strict compiler option and added an option to build tests or not (default off). But now the compile flags are only set for the tests and not for the header interface target. So if someone else uses the interface target to import the compile definitions (C++ 11 and zlib stuff), then they wouldn't import the compile flags anyway.

Does that work with Kratos?

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,69 +7,94 @@
...
+    set( VTU11_TEST_SOURCES 
+         test/main_test.cpp 
+         test/pwrite_pyramids3D_test.cpp 
+         test/utilities_test.cpp 
+         test/vtu11_testing.cpp 
+         test/vtu11_testing.hpp 
+         test/write_hexahedras3D_test.cpp 
+         test/write_icosahedron3D_test.cpp 
+         test/write_pyramids3D_test.cpp 
+         test/write_square2D_test.cpp )
```
Comment (philbucher) [inline CMakeLists.txt:61]: why no `GLOB`?

**Review by philbucher — APPROVED** [review]
> @philbucher I hope the CI stuff still does the same thing as before. Would be nice if you could have a quick look.

seems ok, I checked

> @philbucher Also, I removed the strict compiler option and added an option to build tests or not (default off). But now the strict compile flags are only set for the tests and not for the header interface target. So if someone else uses the interface target to import the compile definitions (C++ 11 and zlib stuff), then they wouldn't import the compile flags anyway.

make sense to me

> Does that work with Kratos?

yes it should

File: CMakeLists.txt (hunk 1 above)
Comment (phmkopp) [reply CMakeLists.txt:61]: In my code I added a `files.cmake` in every source folder which only sets one variable with all file names.

Comment (phmkopp) [reply CMakeLists.txt:61]: Like here: https://gitlab.com/phmkopp/mlhp/-/blob/master/source/core/files.cmake

**Pull request merged**